tools/unitctl/unit-client-rs/src/unitd_docker.rs

```diff
@@ -0,0 +1,403 @@
+use std::collections::HashMap;
+use std::fs::read_to_string;
+use std::path::PathBuf;
+
+use crate::futures::StreamExt;
+use crate::unit_client::UnitClientError;
+use crate::unitd_process::UnitdProcess;
+use bollard::container::{Config, ListContainersOptions, StartContainerOptions};
+use bollard::image::CreateImageOptions;
+use bollard::models::{ContainerCreateResponse, HostConfig, Mount, MountTypeEnum};
+use bollard::secret::ContainerInspectResponse;
+use bollard::{models::ContainerSummary, Docker};
+use regex::Regex;
+use serde::ser::SerializeMap;
+use serde::{Serialize, Serializer};
+
+#[derive(Clone, Debug)]
+pub struct UnitdContainer {
+    pub container_id: Option<String>,
+    pub container_image: String,
+    pub command: Option<String>,
+    pub mounts: HashMap<PathBuf, PathBuf>,
+    pub platform: String,
+    details: Option<ContainerInspectResponse>,
+}
+
+impl From<&ContainerSummary> for UnitdContainer {
+    fn from(ctr: &ContainerSummary) -> Self {
+        // we assume paths from the docker api are absolute
+        // they certainly have to be later...
+        let mut mounts = HashMap::new();
+        if let Some(mts) = &ctr.mounts {
+            for i in mts {
+                if let Some(ref src) = i.source {
+                    if let Some(ref dest) = i.destination {
+                        mounts.insert(PathBuf::from(dest.clone()), PathBuf::from(src.clone()));
+                    }
+                }
+            }
+        }
+
+        UnitdContainer {
+            container_id: ctr.id.clone(),
+            container_image: format!(
+                "{} (docker)",
+                ctr.image.clone().unwrap_or(String::from("unknown container")),
+            ),
+            command: ctr.command.clone(),
+            mounts: mounts,
+            platform: String::from("Docker"),
+            details: None,
+        }
+    }
+}
+
+impl From<&UnitdContainer> for UnitdProcess {
+    fn from(ctr: &UnitdContainer) -> Self {
+        let version = ctr.details.as_ref().and_then(|details| {
+            details.config.as_ref().and_then(|conf| {
+                conf.labels.as_ref().and_then(|labels| {
+                    labels
+                        .get("org.opencontainers.image.version")
+                        .and_then(|version| Some(version.clone()))
+                })
+            })
+        });
+        let command = ctr.command.clone().and_then(|cmd| {
+            Some(format!(
+                "{}{} [{}{}]",
+                "unit: main v",
+                version.or(Some(String::from(""))).unwrap(),
+                ctr.container_image,
+                ctr.rewrite_socket(
+                    cmd.strip_prefix("/usr/local/bin/docker-entrypoint.sh")
```

Comment:
Will this path always be constant?

Reply:
Unless we change our dockerfiles it will be :)